packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -211,12 +212,7 @@ class CpuProfileData {
     }
 
     // Initialize all CPU samples.
-    final stackTraceEvents =
-        (json[traceEventsKey] ?? []).cast<Map<String, dynamic>>();
-    final samples = stackTraceEvents
-        .map((trace) => CpuSampleEvent.parse(trace))
-        .toList()
-        .cast<CpuSampleEvent>();
+    final samples = json.traceEvents ?? [];
```

Comment:
can we give the empty list a type?

Reply:
I'm sort of following ["DON’T write type arguments on generic invocations that are inferred"](https://dart.dev/effective-dart/design#dont-write-type-arguments-on-generic-invocations-that-are-inferred) here. Since `json.traceEvents` is a `List<CpuSampleEvent>?`, the type argument is inferred on `[]`, and `samples` has the correct type of `List<CpuSampleEvent>`.

All that being said though, I'm happy to add a type argument if it makes the code more readable.